sklearn/utils/validation.py

```diff
@@ -1429,6 +1429,12 @@ def column_or_1d(y, *, dtype=None, warn=False):
     warn : bool, default=False
        To control display of warnings.
 
+    device : device, default=None
+        `device` object.
+        See the :ref:`Array API User Guide <array_api>` for more details.
+
+        .. versionadded:: 1.6
```

Comment:
@glemaitre @jeremiedbb should we change this or backport?

Reply:
Let's backport since we have the experimental guardrail